buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -118,175 +139,37 @@ class BuildPlugin implements Plugin<Project> {
-                println "  JDK Version           : ${JavaVersion.toVersion(gradleJavaVersion)} (${gradleJavaVersionDetails})"
-                println "  JAVA_HOME             : ${gradleJavaHome}"
-            }
-            println "  Random Testing Seed   : ${project.testSeed}"
-            println '======================================='
-
-            // enforce Java version
-            if (compilerJavaVersionEnum < minimumCompilerVersion) {
-                final String message =
-                        "the compiler java.home must be set to a JDK installation directory for Java ${minimumCompilerVersion}" +
-                                " but is [${compilerJavaHome}] corresponding to [${compilerJavaVersionEnum}]"
-                throw new GradleException(message)
-            }
-
-            if (runtimeJavaVersionEnum < minimumRuntimeVersion) {
-                final String message =
-                        "the runtime java.home must be set to a JDK installation directory for Java ${minimumRuntimeVersion}" +
-                                " but is [${runtimeJavaHome}] corresponding to [${runtimeJavaVersionEnum}]"
-                throw new GradleException(message)
-            }
-
-            final Map<Integer, String> javaVersions = [:]
-            for (int version = 8; version <= Integer.parseInt(minimumCompilerVersion.majorVersion); version++) {
-                if(System.getenv(getJavaHomeEnvVarName(version.toString())) != null) {
-                    javaVersions.put(version, findJavaHome(version.toString()));
-                }
-            }
-
-            final int numberOfPhysicalCores = numberOfPhysicalCores(project.rootProject)
-            if (javaVersions.isEmpty() == false) {
-
-                ExecutorService exec = Executors.newFixedThreadPool(numberOfPhysicalCores)
-                Set<Future<Void>> results = new HashSet<>()
-
-                javaVersions.entrySet().stream()
-                        .filter { it.getValue() != null }
-                        .forEach { javaVersionEntry ->
-                    results.add(exec.submit {
-                        final String javaHome = javaVersionEntry.getValue()
-                        final int version = javaVersionEntry.getKey()
-                        if (project.file(javaHome).exists() == false) {
-                            throw new GradleException("Invalid JAVA${version}_HOME=${javaHome} location does not exist")
+        // Need to do it here to support external plugins
+        if (project == project.rootProject) {
+            GlobalInfoExtension globalInfo = project.extensions.getByType(GlobalInfoExtension)
+
+            // wait until global info is populated because we don't know if we are running in a fips jvm until execution time
+            globalInfo.ready {
+                project.subprojects { Project subproject ->
+                    ExtraPropertiesExtension ext = subproject.extensions.getByType(ExtraPropertiesExtension)
```

Comment:
Why aren't we storing `inFipsJvm` in the `GlobalInfoExtension` ? Would be nice to stop using extra properties here.

Reply:
That is a potential solution #42042. In any case, ditching all these extra properties is to be addressed by that issue.